### PR TITLE
dev-libs/hyperscan: disable `-Werror` by saying that Gentoo build type is production-ready (Release)

### PR DESCRIPTION
**Pull request by podshumok**

### CODE REVIEW

**Comment by juippis** [comment]
Ehh, isnt that for upstream to decide? Is there a bug created for upstream about this?

**Comment by podshumok** [comment]
Upstream uses '-Werror` for all non-production builds.
Production build types (i.e. CMake build types) are specified via regex. Sure, they don't have any idea about Gentoo build type in CMake, so they consider this build type as non-release/non-production.
Few previous bug fixes were all about disabling different `-W` options because of active `-Werror`

File: dev-libs/hyperscan/hyperscan-5.1.1.ebuild [hunk 1]
```diff
@@ -24,16 +24,13 @@ BDEPEND="${RDEPEND}"
 
 REQUIRED_USE="cpu_flags_x86_ssse3 ${PYTHON_REQUIRED_USE}"
 
-src_prepare() {
-	# upstream workaround
-	append-cxxflags -Wno-redundant-move
-	cmake-utils_src_prepare
-}
+PATCHES=( "${FILESDIR}/gentoo_build_is_release.patch" )
```
Comment (mgorny) [inline dev-libs/hyperscan/hyperscan-5.1.1.ebuild]: Wouldn't it actually be better to use a sane build type than patch upstream build system to accept historical Gentoo insanity?

Comment (podshumok) [reply dev-libs/hyperscan/hyperscan-5.1.1.ebuild]: And override CFLAGS with "sane defaults"?

Comment (mgorny) [reply dev-libs/hyperscan/hyperscan-5.1.1.ebuild]: We override the CFLAGS override.

Comment (podshumok) [reply dev-libs/hyperscan/hyperscan-5.1.1.ebuild]: with custom build type?

Comment (mgorny) [reply dev-libs/hyperscan/hyperscan-5.1.1.ebuild]: With CMake override for all standard build types.

Comment (podshumok) [reply dev-libs/hyperscan/hyperscan-5.1.1.ebuild]: Docs on `cmake-utils` eclass don't contain anything about it

**Comment by arkamar** [comment]
This issue is fixed in ebuilds present in the overlay. This PR is obsolete and should be closed.

**Pull request closed without merging**